### PR TITLE
Implement showing medicine frequency labels depending on the country - Final

### PR DESCRIPTION
**Pull request by janhavisinghh**
https://app.clubhouse.io/simpledotorg/story/4759/implement-showing-medicine-frequency-labels-depending-on-the-country-in-medicinefrequencysheet-and-teleconsultmedicinesview

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/teleconsultlog/medicinefrequency/MedicineFrequencyModel.kt [hunk 1]
```diff
@@ -2,15 +2,20 @@ package org.simple.clinic.teleconsultlog.medicinefrequency
...
-    val medicineFrequency: MedicineFrequency
+    val medicineFrequency: MedicineFrequency,
+    val medicineFrequencyToFrequencyChoiceItemMap: Map<MedicineFrequency?, DrugFrequencyChoiceItem>
 ) : Parcelable {
 
   companion object {
-    fun create(medicineFrequency: MedicineFrequency) = MedicineFrequencyModel(
-        medicineFrequency = medicineFrequency
+    fun create(
+        medicineFrequency: MedicineFrequency, medicineFrequencyToFrequencyChoiceItemMap: Map<MedicineFrequency?, DrugFrequencyChoiceItem>
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/teleconsultlog/medicinefrequency/MedicineFrequencyModel.kt:15]: Can we format the second parameter in the next line?

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks good. Few unit tests are failing and left a few comments. PTAL?

File: app/src/main/java/org/simple/clinic/teleconsultlog/prescription/medicines/TeleconsultMedicineItem.kt [hunk 2]
```diff
@@ -51,7 +51,8 @@ data class TeleconsultMedicineItem(
     )
 
     val frequency = prescribedDrug.frequency ?: defaultFrequency
-    binding.medicineFrequencyButton.text = frequency.toString()
+    val frequencyLabel = medicineFrequencyToFrequencyChoiceItemMap[frequency]!!.label
+    binding.medicineFrequencyButton.text = frequencyLabel
```
[the remark this answers is not in the document]
Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/teleconsultlog/prescription/medicines/TeleconsultMedicineItem.kt:55]: We are using `DrugFrequencyFactoryModule` to load `drugFrequencyChoiceItems` based on the country the user is in. Also, we are only loading different drug frequency labels based on the country(like `PID` for `Ethiopia`) we are saving drug frequencies based on the common acronyms for it(i.e. BD, TDS, QDS, etc)

**Comment by Honey14** [comment]
Cannot test this because of [this](https://simpledotorg.slack.com/archives/CFK2PHJ1L/p1631692063251600) issue. Removing myself as a reviewer for now.

**Review by Honey14 — CHANGES_REQUESTED** [review]
LGTM. Frequency labels are not changing for Ethiopia in teleconsult view. PTAL

**Review by Honey14 — APPROVED** [review]
LGTM

**Pull request merged**